scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -1584,82 +1674,37 @@ function Validate-GatewaySettings {
 }
 
 function Get-DpsProvisioningSettings {
-    $idCertFilesProvided = $false
-    $attestationMethod = 'tpm' # default
-    if ($SymmetricKey) {
+    $attestationMethod = ''
+    if ($DpsTpm) {
```

Comment:
These won't be set if the user calls `Initialize-` directly and doesn't pass any switches, just the required parameters.

One way is you pass the `$PSCmdlet.ParameterSetName` value down from `Initialize-` to here (via `Set-ProvisioningMode`) as a parameter.

The easier way is to add a `switch ($PSCmdlet.ParameterSetName)` in `Initialize-` that manually sets the corresponding switch variable to `$true`. I prefer this one.

Reply:
Thanks I didn't understand this when you described this earlier but I see your point. I have fixed this now.